include/fastcdr/xcdr/optional.hpp

```diff
@@ -0,0 +1,400 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef _FASTCDR_XCDR_OPTIONAL_HPP_
+#define _FASTCDR_XCDR_OPTIONAL_HPP_
+
+#include <new>
+#include <type_traits>
+
+#include "../exceptions/BadOptionalAccessException.hpp"
+
+namespace eprosima {
+namespace fastcdr {
+
+namespace detail {
+template<class T, class E = void>
+struct optional_storage
+{
+    union
+    {
+        char dummy_; T val_;
```

Comment:
eProsima's code style encourages defining one variable per line to help readability.

I suppose the dummy variable is defined because

Reply:
Done in e3a6645